### PR TITLE
Smart charging add a select button to switch from kW to amps in the charging and site curve

### PR DESCRIPTION
**Pull request by OliveGerste**

### CODE REVIEW

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts [hunk 1]
```diff
@@ -27,10 +36,16 @@ export class ChargingStationSmartChargingLimitPlannerChartComponent implements O
...
 
-  @ViewChild('primary', {static: true}) public primaryElement!: ElementRef;
+  @ViewChild('primary', { static: true }) public primaryElement!: ElementRef;
   @ViewChild('danger', { static: true }) public dangerElement!: ElementRef;
-  @ViewChild('chart', {static: true}) public chartElement!: ElementRef;
+  @ViewChild('chart', { static: true }) public chartElement!: ElementRef;
+
+  public selectedUnit = ConsumptionUnit.KILOWATT;
 
+  public unitMap = [
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts:45]: It's used in more than one file, it should be in /shared

File: src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.ts [hunk 2]
```diff
@@ -64,6 +76,12 @@ export class SiteAreaConsumptionChartComponent implements OnInit, AfterViewInit
       Validators.required,
     ]));
     this.dateControl.setValue(this.startDate);
+    this.centralServerService.getSiteArea(this.siteAreaId)
+    .subscribe((siteArea) => {
+      this.siteArea = siteArea;
+    }, (error) => {
+      this.messageService.showErrorMessage('transactions.graph.error');
```
Comment (mahdibenromdhane) [inline src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.ts:83]: It should be a getSiteArea error

File: src/app/shared/component/consumption-chart/consumption-chart.component.ts [hunk 3]
```diff
@@ -79,19 +97,37 @@ export class ConsumptionChartComponent implements AfterViewInit {
...
       });
   }
 
+  public getCharger() {
+    if (this.transaction) {
+      this.centralServerService.getChargingStation(this.transaction.chargeBoxID)
+      .subscribe((charger) => {
+        this.charger = charger;
+      }, (error) => {
+        this.messageService.showErrorMessage('transactions.graph.error');
```
Comment (mahdibenromdhane) [inline src/app/shared/component/consumption-chart/consumption-chart.component.ts:115]: It should be a getChargingStation error

**Comment by mahdibenromdhane** [comment]
image
- if you click on the radio and checkbox once , the hover style do not disappear. 

image
- An error occur with some of the transaction and also a spinner need to be added so a user can't click several time as shown in the screenshot.

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts (hunk 1 above)
Comment (OliveGerste) [reply src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts:45]: How do you mean this. Where and how would you store the Array?

**Comment by OliveGerste** [comment]
> image
> 
> * if you click on the radio and checkbox once , the hover style do not disappear.
> 
> image
> 
> * An error occur with some of the transaction and also a spinner need to be added so a user can't click several time as shown in the screenshot.

1. I think, this is a bug of Angular on localhost. I have it everywhere in the Application. 

2. I cannot reproduce the bug with this session and do not understand what really happens there. Maybe we could have a talk tomorrow.

**Pull request closed without merging**